File: projects/punk0/code/sort_desc.py
# sort the list in descending order
#   [1,4,2,3]
# returns:
#   [4,3,2,1]
def sort_desc(x):
    # we can use the sorted() function
    # make a sorted copy

    # x.sort method mutates the list so
    # first we need to copy it and then
    # sort the copy
    r = []
    for v in x:
        r.append(v)

    r.sort(reverse=True)

    return r


print(sort_desc([1, 1, 2, 3, 3, 4, 1, 2, 7, 9]))

File: projects/punk0/code/sort_desc.js
// sort the list in descending order
//   [1,4,2,3]
// returns:
//   [4,3,2,1]
function sort_desc(x) {
  // x.sort() mutates the array itsef
  // so we will copy it and sort the
  // copy
  let r = [];
  for (let v of x) {
    r.push(v);
  }

  r.sort();
  r.reverse()

  return r;
}

console.log(JSON.stringify(sort_desc([1, 1, 2, 3, 3, 4, 1, 2, 7, 9])));
